oracle-plugin/src/e2e-test/java/io.cdap.plugin/OracleClient.java

```diff
@@ -74,6 +74,122 @@ public static void createTargetTable(String targetTable, String schema) throws S
     }
   }
 
+  public static void createSourceDatatypesTable(String sourceTable, String schema) throws SQLException,
+    ClassNotFoundException {
+    try (Connection connect = getOracleConnection(); Statement statement = connect.createStatement()) {
+      String createSourceTableQuery2 = "CREATE TABLE " + schema + "." + sourceTable +
+        "(ID VARCHAR2(100) PRIMARY KEY, COL1 CHAR, COL2 CHAR(10), COL3 VARCHAR(3), COL4 VARCHAR2(3)," +
+        "COL5 NCHAR, COL6 NCHAR(12), COL7 NVARCHAR2(12), COL8 CLOB, COL9 NCLOB, COL10 LONG, COL11 ROWID," +
+        " COL12 NUMBER(4), COL13 NUMBER(*), COL14 NUMBER(*,2), COL15 NUMBER(10,-3), COL16 NUMBER, COL17 DECIMAL(4), " +
+        "COL18 DECIMAL(*), COL19 DECIMAL(*,2), COL20 DECIMAL(10,-3), COL21 DECIMAL, COL22 FLOAT, COL23 FLOAT(4), " +
+        "COL24 INTEGER, COL25 DOUBLE PRECISION, COL26 REAL, COL27 SMALLINT, COL28 TIMESTAMP, COL29 TIMESTAMP(9)," +
+        " COL30 TIMESTAMP WITH TIME ZONE, COL31 INTERVAL DAY(6) TO SECOND (5), COL32 INTERVAL YEAR(4) TO MONTH," +
+        " COL33 DATE, COL34 BINARY_FLOAT, COL35 BINARY_DOUBLE)";
+      statement.executeUpdate(createSourceTableQuery2);
+
+      // Insert dummy data.
+      statement.executeUpdate("INSERT INTO " + schema + "." + sourceTable + " (ID, COL1, COL2, COL3, COL4,COL5," +
```

Comment:
2 comments 
1) Can this be extracted out in a config or something similar? This would ensure any further change might not require an explicit code change but a config change instead. 
2) If not, Can this be made a reusable piece as you may have to repeat it for many other relational database type (assuming most data types are similar).

Reply:
Removed all these database/table specific details from the OracleClient file and moved them to pluginParameters.properties file. Any further change now in the database or the type of tables used will require modifying the properties file only